samples/matter/common/cmake/source_common.cmake

```diff
@@ -0,0 +1,53 @@
+#
+# Copyright (c) 2024 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+# NORDIC SDK APP START
+target_include_directories(app PRIVATE
+    src
+    ${COMMON_ROOT}/src
+    ${NLIO_ROOT}/include
+    ${ZEPHYR_CONNECTEDHOMEIP_MODULE_DIR}/zzz_generated/app-common
+)
+
+# Set sources that are used by all samples
+target_sources(app PRIVATE
+    src/zap-generated/IMClusterCommandHandler.cpp
+    src/zap-generated/callback-stub.cpp
+    ${COMMON_ROOT}/src/board/led_widget.cpp
+    ${COMMON_ROOT}/src/board/board.cpp
+    ${COMMON_ROOT}/src/app/task_executor.cpp
+    ${COMMON_ROOT}/src/app/matter_init.cpp
+    ${COMMON_ROOT}/src/app/matter_event_handler.cpp
+)
+
+
+# Set specific source that depends on Kconfigs
+if(CONFIG_CHIP_OTA_REQUESTOR OR CONFIG_MCUMGR_TRANSPORT_BT)
+    target_sources(app PRIVATE ${COMMON_ROOT}/src/dfu/ota/ota_util.cpp)
+endif()
+
+if(CONFIG_MCUMGR_TRANSPORT_BT)
+    zephyr_library_link_libraries(MCUBOOT_BOOTUTIL)
+    target_sources(app PRIVATE ${COMMON_ROOT}/src/dfu/smp/dfu_over_smp.cpp)
+endif()
+
+if(CONFIG_NCS_SAMPLE_MATTER_OPERATIONAL_KEYS_MIGRATION_TO_ITS)
+    target_sources(app PRIVATE ${COMMON_ROOT}/src/migration/migration_manager.cpp)
+endif()
+
+if(CONFIG_CHIP_NUS)
+    target_sources(app PRIVATE ${COMMON_ROOT}/src/bt_nus/bt_nus_service.cpp)
+endif()
+
+if(CONFIG_NCS_SAMPLE_MATTER_DIAGNOSTIC_LOGS)
+    target_sources(app PRIVATE ${COMMON_ROOT}/src/diagnostic/diagnostic_logs_provider.cpp)
+if(CONFIG_NCS_SAMPLE_MATTER_DIAGNOSTIC_LOGS_CRASH_LOGS)
+    # Wrap z_fatal_error to allow injecting crash data into the retention memory.
+    target_link_options(app INTERFACE -Wl,--wrap=z_fatal_error)
+    target_sources(app PRIVATE ${COMMON_ROOT}/src/persistent_storage/persistent_storage_util.cpp)
+    target_sources(app PRIVATE ${COMMON_ROOT}/src/diagnostic/diagnostic_logs_crash.cpp)
+endif()
+endif()
```

Comment:
This part is not available yet and shouldn't be here :)

Reply:
Okey, should I leave it in any CmakeLists file?

Reply:
This part is prepared only for the TestEvent and will be merged to the main branch after NCS 2.6.0 release, so currently, there are no such entries in any cmakelists files.